src/MapWindow/MapWindowTraffic.cpp

```diff
@@ -116,6 +116,93 @@ MapWindow::DrawFLARMTraffic(Canvas &canvas,
   }
 }
 
+
+/**
+ * Draws the GliderLink traffic icons onto the given canvas
+ * @param canvas Canvas for drawing
+ */
+void
+MapWindow::DrawGLinkTraffic(Canvas &canvas,
+                            const PixelPoint aircraft_pos) const
+{
+  // Return if FLARM icons on moving map are disabled
+  if (!GetMapSettings().show_flarm_on_map)
+    return;
+
+  const GliderLinkTrafficList &traffic = Basic().glink_data.traffic;
+  if (traffic.IsEmpty())
+    return;
+
+  const MoreData &basic = Basic();
+
+  const WindowProjection &projection = render_projection;
+
+  canvas.Select(*traffic_look.font);
+
+  // Circle through the GliderLink targets
+  for (const auto &traf : traffic.list) {
+
+    // Save the location of the target
+    GeoPoint target_loc = traf.location;
+
+    // Points for the screen coordinates for the icon, name and average climb
+    PixelPoint sc, sc_name, sc_av, sc_alt;
+
+    // If FLARM target not on the screen, move to the next one
+    if (!projection.GeoToScreenIfVisible(target_loc, sc))
+      continue;
+
+    // Draw the callsign above the icon
+    sc_name = sc;
+    sc_name.x -= Layout::Scale(10);
```

Comment:
Does this scale properly with all font sizes? Your formula doesn't consider font size at all.

Reply:
The implementation has been copied from FLARM